pkg/ebpf/bytecode/elf.go

```diff
@@ -0,0 +1,95 @@
+// +build linux_bpf
+
+package bytecode
+
+import (
+	"fmt"
+	"os"
+	"path"
+
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/DataDog/ebpf/manager"
+	gore "github.com/lebauce/go-re"
+)
+
+var defaultCompilationFlags = []string{
+	"-DRUNTIME_COMPILATION=1",
+	"-DKBUILD_MODNAME=\"foo\"",
+	"-D__KERNEL__",
+	"-DCONFIG_64BIT",
+	"-D__BPF_TRACING__",
+	"-Wno-unused-value",
+	"-Wno-pointer-sign",
+	"-Wno-compare-distinct-pointer-types",
+	"-Wunused",
+	"-Wall",
+	"-Werror",
+}
+
+// Options consolidates all configuration params associated to the eBPF byte code management
+type Options struct {
+	BPFDir               string
+	Debug                bool
+	EnableIPv6           bool
+	OffsetGuessThreshold uint64
+}
+
+// GetNetworkTracerELF obtains the eBPF bytecode used by the Network Tracer.
+// First, it attempts to compile the eBPF bytecode on the fly, using the host kernel
+// headers and the bundled compiler.  If this process fail for some reason
+// (eg. system headers not available), we fall back to the pre-compiled ELF file
+// that relies on offset guessing.
+func GetNetworkTracerELF(opts Options) (AssetReader, []manager.ConstantEditor, error) {
+	compiler := gore.NewEBPFCompiler(true)
+	defer compiler.Close()
+
+	var (
+		in  = path.Join(opts.BPFDir, "tracer-ebpf.c")
```

Comment:
This will be incorrect if the C file is bundled within the binary (with `ebpf_bindata` build flag).

Reply:
Indeed. Please use `bytecode.GetReader("tracer-ebpf.c")`

Reply:
Thank you for pointing that out.
Per our chat in slack I'll leave it as it is for now for the following reasons:
* This would require some non-trivial amount of work to get the header inclusion working;
* It only impacts the development workflow;
* If the compilation fails the error is handled gracefully and we use the bundled pre-compiled ELF file;

In addition to that I'm not even sure if this PR will ever get merged ;-) If this turns out to be a real issue I'm definitely happy to work on making this compatible with the `ebpf_bindata` buildflag